### ISSUE TITLE
How to obtain high registration accuracy(i.e. small LDE) with only Dirlab dataset?

### ISSUE THREAD

**Issue by XXZhou25** [Timmy-Fang/Deformable-Image-Registration-Projects#4]
Hi, 
I am desperate with my model training. 
I used basic Voxelmorph model, and my loss is local NCC + l2 norm gradient of dvf, 
I've tried almost all possible learning rate, regularizor weight, my LDEs are almost not changed.
I mean almost the same as original LDEs.

I tried to use only dirlab, and I even enlarged my dataset to about 400+ CTs. 
No way works so far.
I thought one possible reason could be lack of training data. 

But I am wondering if you only use Dirlab for both training and testing? 
And did you apply data augmentation to enrich your training? 
Your paper said you only used Dirlab for all procedures, I just want to double check.

Also, I am using random pairs as input, which means 2 input images may not come from one patient, and can be random phases.

**Comment by XXZhou25** [comment]
And I am also wondering why in your LDE computation, you are deducting 1 from each landmark? 
`
mov_lmk0 = (mov_lmk-1) * resize_factor0 + delta0ref_lmk0 = (ref_lmk-1) * resize_factor0 + delta0`

**Comment by XXZhou25** [comment]
And also from your code in MIR 3D, it looks like you didn't apply gradient loss? I mean a regularizer. 
It seems you are using NCC + MSE loss instead. 
Then may I know how you 'guarantee' the smoothness of deformation field? ('guarantee' I mean of course not 100% smooth, but at least penalize unsmoothness).

**Comment by YingYuxinLeah** [comment]
> And also from your code in MIR 3D, it looks like you didn't apply gradient loss? I mean a regularizer. It seems you are using NCC + MSE loss instead. Then may I know how you 'guarantee' the smoothness of deformation field? ('guarantee' I mean of course not 100% smooth, but at least penalize unsmoothness).

Do you know how to train the model now?The loss doesn't drop and I don't know why.
